Refactor

ninth-dev opened the pull request:
- Add `view_id` to `BufferContent` and `FileContent` to simplify focus_view
- Merge `add_window_views` and `add_group_views`

ninth-dev committed:
Refactor

- Add view_id to BufferContent and FileContent to simplify focus_view
- Merge add_window_views and add_group_views

ssahayam-zd commented on opentabs.py:
```diff
@@ -199,34 +182,11 @@ def create_file_panel_item(self, some_content):
-        return view
-
-    return None
+      window.focus_view(some_content)
```
How does this work? `tracked_views` has `FileContents` or `BufferContents` not a view? Does something convert it into the view?

ninth-dev replied:
content will have `view` instead of just `view_id`.

ssahayam-zd commented on opentabs.py:
```diff
@@ -6,7 +6,8 @@ class FileContents:
   project_folder_path = "[project]"
 
-  def __init__(self, file_name, short_name, folder_name):
+  def __init__(self, view_id, file_name, short_name, folder_name):
```
can we add the `view_id` to `__str__` as well?

ssahayam-zd commented on opentabs.py:
```diff
@@ -138,48 +137,32 @@ def run(self):
 
       for view in views:
+
+        self.index += 1 #start at 0
```
this whole block is kind of meaningless if you have 1 group, which is why I separated out the methods for creating views in my version

ninth-dev replied:
wrapped the code in a if 

```python
if not self.has_group:
  ....
```

ssahayam-zd commented on opentabs.py:
```diff
@@ -112,13 +114,10 @@ def run(self):
     self.buffers = []
```
I think we can remove `buffers`

ninth-dev committed:
More Refactor

- use view over view_id
- remove buffer as it is no longer required
- move some logic into the FileContents and BufferContents

ssanj reviewed (approved):
Thanks for your contributions!

![](https://media.giphy.com/media/N5VZscMsgpKbbAwB30/giphy.gif)
